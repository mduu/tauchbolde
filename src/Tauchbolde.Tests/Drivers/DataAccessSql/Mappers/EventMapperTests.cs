using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Newtonsoft.Json;
using Tauchbolde.Driver.DataAccessSql.DataEntities;
using Tauchbolde.Driver.DataAccessSql.Mappers;
using Xunit;
using EventData = Tauchbolde.Driver.DataAccessSql.DataEntities.EventData;

namespace Tauchbolde.Tests.Drivers.DataAccessSql.Mappers
{
    [UseReporter(typeof(DiffReporter))]
    public class EventMapperTests
    {
        [Fact]
        public void Map()
        {
            var dataObject = new EventData
            {
                Id = new Guid("CF0CFA4E-DA42-4702-ACAA-5FCDC68E4FCE"),
                Name = "Test Event",
                Description = "A description",
                Canceled = true,
                Deleted = true,
                Location = "The location",
                MeetingPoint = "The Meeting-Point",
                OrganisatorId = new Guid("F82BE2F0-EE77-4C61-AFCF-8F604E133E0B"),
                StartTime = new DateTime(2019, 8, 1, 19, 0, 0),
                EndTime = new DateTime(2019, 8, 2, 12, 30, 0),
                Participants = Enumerable.Empty<ParticipantData>().ToList(),
                Comments = Enumerable.Empty<CommentData>().ToList(),
            };

            var domainObject = dataObject.MapTo();
            var newDataObject = domainObject.MapTo();
            
            newDataObject.Should().BeEquivalentTo(dataObject);
            Approvals.VerifyJson(JsonConvert.SerializeObject(
                new
                {
                    dataObject,
                    domainObject,
                    newDataObject
                }));
        }
    }
}