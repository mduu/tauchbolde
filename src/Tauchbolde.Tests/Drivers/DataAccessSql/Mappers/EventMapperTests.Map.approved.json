{
    "dataObject": {
        "Name": "Test Event",
        "OrganisatorId": "f82be2f0-ee77-4c61-afcf-8f604e133e0b",
        "Organisator": null,
        "Location": "The location",
        "MeetingPoint": "The Meeting-Point",
        "Description": "A description",
        "StartTime": "2019-08-01T19:00:00",
        "EndTime": "2019-08-02T12:30:00",
        "Canceled": true,
        "Deleted": true,
        "Participants": [],
        "Comments": [],
        "StartEndTimeAsString": "01.08.2019 - 02.08.2019",
        "Id": "cf0cfa4e-da42-4702-acaa-5fcdc68e4fce"
    },
    "domainObject": {
        "Name": "Test Event",
        "OrganisatorId": "f82be2f0-ee77-4c61-afcf-8f604e133e0b",
        "Organisator": null,
        "Location": "The location",
        "MeetingPoint": "The Meeting-Point",
        "Description": "A description",
        "StartTime": "2019-08-01T19:00:00",
        "EndTime": "2019-08-02T12:30:00",
        "Canceled": true,
        "Deleted": true,
        "Participants": [],
        "Comments": [],
        "StartEndTimeAsString": "01.08.2019 - 02.08.2019",
        "Id": "cf0cfa4e-da42-4702-acaa-5fcdc68e4fce"
    },
    "newDataObject": {
        "Name": "Test Event",
        "OrganisatorId": "f82be2f0-ee77-4c61-afcf-8f604e133e0b",
        "Organisator": null,
        "Location": "The location",
        "MeetingPoint": "The Meeting-Point",
        "Description": "A description",
        "StartTime": "2019-08-01T19:00:00",
        "EndTime": "2019-08-02T12:30:00",
        "Canceled": true,
        "Deleted": true,
        "Participants": [],
        "Comments": [],
        "StartEndTimeAsString": "01.08.2019 - 02.08.2019",
        "Id": "cf0cfa4e-da42-4702-acaa-5fcdc68e4fce"
    }
}